### DISCUSSION TITLE
V3 Release Roadmap

### DISCUSSION THREAD

**Discussion by simonmcallister0210** [simonmcallister0210/cognito-srp-helper#49] [Ideas]
A roadmap of what to expect in the version 3 release of this project. If you have any ideas on what to add feel free to open a new discussion

### Here's a list of what I think _should_ be included so far:
- Move away from crypto-js to standard crypto libraries in browser and NodeJS
- Make the interface less opinionated, e.g. no wrapping functions, no user/device specific SRP, no AWS-SDK specific typing
- Testing across all browsers
- Testing across [maintained, active, and current](https://nodejs.org/en/about/previous-releases) versions of NodeJS

### Here's a list of what I think would be nice to have:
- Performance testing to spot degradations in performance
- Bundle size metrics to avoid bundle size creeping up too much
- Examples sub-folder. Not only to provide useful demos, but to also help spot issues with TS types, imports, etc.

**Comment by DmitryLyubetsky** [comment]
Hello @simonmcallister0210 

We’re migrating our SRP flow, including the DEVICE_PASSWORD_VERIFIER challenge. Your library handles this case when a user’s device sends authentication requests directly to AWS Cognito and does not transmit the user’s password. The library heavily relies on the srpSession object, which includes the user’s password.

However, in our architecture, the backend acts as a proxy between user’s device and AWS Cognito. Our goal is to avoid sending the user’s password to the backend while still being able to complete the SRP flow and sign data using your library’s functionality. We’re considering creating the srpSession object on the frontend and sending it to the backend, but this approach seems problematic due to the inclusion of the password.

We’re seeking your advice on how to proceed. Perhaps there’s a way to create an srpSession object without the password and still use your library’s wrapper functions. Or, maybe there are future plans to incorporate zero-knowledge principles into the library?
Any guidance or suggestions would be greatly appreciated.
